xds/internal/client/rds.go

```diff
@@ -137,9 +137,13 @@ func getClusterFromRouteConfiguration(rc *xdspb.RouteConfiguration, target strin
 				continue
 			}
 			dr := vh.Routes[len(vh.Routes)-1]
-			if dr.GetMatch() == nil && dr.GetRoute() != nil {
-				return dr.GetRoute().GetCluster()
+			if dr.GetMatch() == nil || dr.GetRoute() == nil {
```

Comment:
Nit: if `match` is moved to the same condition, you can save a `GetMatch()` call.

```go
if match := dr.GetMatch(); match == nil || match.GetPrefix() != "" {
  continue
}
if dr.GetRoute() == nil {
  continue
}
return dr.GetRoute().GetCluster()
```